tools/bpf2c/bpf_code_generator.h

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+#include "ebpf.h"
+#include "ebpf_structs.h"
+#pragma warning(push)
+#pragma warning(disable : 4458) /* declaration of 'name' hides class member */
+#include "elfio/elfio.hpp"
+#pragma warning(pop)
+
+class bpf_code_generator
+{
+  public:
+    /**
+     * @brief Construct a new bpf code generator object.
+     *
+     * @param[in] path Path to the eBPF file to parse.
+     * @param[in] c_name C compatible name to export this as.
+     */
+    bpf_code_generator(const std::string& path, const std::string& c_name);
+
+    /**
+     * @brief Construct a new bpf code generator object from raw eBPF byte code.
+     *
+     * @param[in] c_name C compatible name to export this as.
+     * @param[in] instructions Set of eBPF instructions to use.
+     */
+    bpf_code_generator(const std::string& c_name, const std::vector<ebpf_inst>& instructions);
```

Comment:
In case of raw bytecode, it is possible that the ebpf program is referencing a map fd, and that map was created by user mode app. How will the resultant C code contain the map definition of the map referenced in the program?

Reply:
Raw byte code support was added solely for testing. The flow is ELF -> .sys.